Create nodes for media type (#285)

mwoelk opened the pull request:
This pull request adds support to extract attributes of type `media` into own nodes.
They are available as `STRAPI__MEDIA` (double undersocre to prevent conflicts with possible collections named media).
This way fragments can be used in gatsby to reduce gql duplication and to keep e.g. the query close to an image component.

Example definition of a fragment in a gatsby component:
```js
export const query = graphql`
    fragment StrapiImage on STRAPI__MEDIA {
        id
        alternativeText
        localFile {
            childImageSharp {
                gatsbyImageData
            }
        }
    }
`;
```

Example usage of the fragment in another gatsby component (assuming there is a collection `page` with a media attribute `image`):
```js
export const query = graphql`
  query($id: String) {
    strapiPage(id: { eq: $id }) {
      title
      locale
      image {
        ...StrapiImage
      }
    }
  }
`
```

strapi-cla commented:
[![CLA assistant check](https://cla.strapi.io/pull/badge/not_signed)](https://cla.strapi.io/strapi/gatsby-source-strapi?pullRequest=286) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.strapi.io/strapi/gatsby-source-strapi?pullRequest=286) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.strapi.io/check/strapi/gatsby-source-strapi?pullRequest=286) it.</sub>

remidej commented:
I think this is a good idea. I'll review soon so we can ship it before the stable release

markkaylor reviewed (changes requested):
- Not sure I understand why this creates `childStrapiMedia` and `childrenStrapiMedia` on each Article entity for example (see screenshot below). These fields don't seem necessary for your described usage. Could you remove them or explain why they are necessary?
- This only handles a single media, but it also possible to have multiple media. You will need to handle this case. You can look at how it is done for [relations](https://github.com/strapi/gatsby-source-strapi/blob/74e9f5c0e38149e4c3c144449312343c1ae4a565/src/normalize.js#L152)

<img width="251" alt="Screenshot 2022-02-24 at 15 33 53" src="https://user-images.githubusercontent.com/26598053/155549283-cfd1538d-cb0b-49aa-bcd4-d21c6b09d3a1.png">
2

mwoelk commented:
Hey, sorry for the delay. Just updated the code and removed the child-field that was created. Also I applied the same logic as with relations if multiple images are used. Thanks for the review :)

markkaylor commented:
Thanks @mwoelk, I'm still seeing the `childStrapiMedia` and `childrenStrapiMedia` nodes. I have your latest commit, and I tried clearing the cache in the gatsby project. 

<img width="203" alt="Screenshot 2022-03-11 at 09 23 40" src="https://user-images.githubusercontent.com/26598053/157830076-e9cdbe38-f33f-4e6c-9b85-aa4049200aec.png">

mwoelk commented:
Hey @markkaylor,
I just tried to reproduce it but with my example it works fine (using both single and multiple images).

![Screenshot 2022-03-11 at 11 56 32](https://user-images.githubusercontent.com/3381449/157854365-5896b679-df91-4e37-93f3-a455fe8da075.png)

Are you sure you've tested it on the latest commit? The error came through this in the first commit which I removed in the second commit. So it shouldn't be in there anymore :)

https://github.com/mwoelk/gatsby-source-strapi/blob/53f7e52fa466ca89a82dff3e73e6dd09baee4103/src/normalize.js#L290

markkaylor commented:
Hey @mwoelk, my bad, I forgot to rebuild the plugin 🤦‍♂️ , the nodes look good. Just a few more things to test and I'll get back to you asap

markkaylor reviewed (approved):
Looks good to me! @remidej @soupette
